### DISCUSSION TITLE
Ways of loading MFE in master app (Conventional routing)

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#4705] [Q&A]
Hi there,

Currently i have this simple MFE setup in master app
![image](https://github.com/web-infra-dev/modern.js/assets/52765572/3de4d4a0-f1ef-4e7a-885d-faafaee9466d)

From the example it looks like it iterate through apps and render them through <Route /> Component. Is it possible to select the sub app from apps explicitly and render them like  <Mfe /> ?

![image](https://github.com/web-infra-dev/modern.js/assets/52765572/10315a08-3eac-4a2e-8468-8db77c22e1dd)

Trying to do 
![image](https://github.com/web-infra-dev/modern.js/assets/52765572/277f0cf2-fbc3-4915-a328-9d159e42f98a)

**Comment by AsuraKev** [comment]
I have a use case where I might need to pass some data to the MFE when its loaded. Coming from Module Federation I was able to do 

import Mfe from 'Mfe';

<Mfe {props} />

Which gives granule control. Not sure when Module federation is implemented into rspack, it ll have a different way of implementing microfrontend in modern js framework?

**Comment by chenjiahan** [comment]
> Not sure when Module federation is implemented into rspack, it ll have a different way of implementing microfrontend in modern js framework?

There is an issue tracking the status of module federation support in Rspack: https://github.com/web-infra-dev/rspack/issues/1930

Although Rspack does not support module federation yet, I think it will be implemented in a new months, then we will re-consider how to provide better MFE support in Modern.js.
